lib/glific/providers/gupshup/partner/partner_api.ex

```diff
@@ -0,0 +1,148 @@
+defmodule Glific.Providers.Gupshup.PartnerAPI do
+  @moduledoc """
+  A module to handle fetching tier related information like quality rating and app rating using partner API
+  """
+
+  alias Glific.Caches
+  alias Plug.Conn.Query
+
+  use Tesla
+  plug(Tesla.Middleware.Logger)
+
+  plug(Tesla.Middleware.FormUrlencoded,
+    encode: &Query.encode/1
+  )
+
+  @partner_url "https://partner.gupshup.io/partner/account"
+  @app_url "https://partner.gupshup.io/partner/app/"
+  @global_organization_id 0
+
+  # fetches partner token
+  @spec get_partner_token :: {:ok, map()} | {:error, any}
+  defp get_partner_token do
+    email = Application.fetch_env!(:glific, :gupshup_partner_email)
+    password = Application.fetch_env!(:glific, :gupshup_partner_password)
+    url = @partner_url <> "/login"
+
+    # Using Cachex.get instead of Caches.get as token expire after 24hrs and we dont want to refresh the cache
+    {:ok, partner_token} = Cachex.get(:glific_cache, {@global_organization_id, "partner_token"})
+
+    if partner_token,
+      do: {:ok, %{partner_token: partner_token}},
+      else: do_get_partner_token(email, password, url)
+  end
+
+  @spec do_get_partner_token(String.t(), String.t(), String.t()) :: {:ok, map()} | {:error, any}
+  defp do_get_partner_token(email, password, url) do
+    post(url, %{"email" => email, "password" => password}, headers: [])
+    |> case do
+      {:ok, %Tesla.Env{status: 200, body: body}} ->
+        {:ok, partner_token} =
+          Jason.decode!(body)
+          |> then(&Caches.set(@global_organization_id, "partner_token", &1["token"]))
+
+        {:ok, %{partner_token: partner_token}}
+
+      {_status, response} ->
+        {:error, "invalid response #{inspect(response)}"}
+    end
+  end
+
+  # fetches app id from phone using partner API
+  @spec get_apps_details(String.t()) :: {:ok, String.t()} | {:error, any}
+  defp get_apps_details(phone) do
+    with {:ok, app_list} <- Caches.get(@global_organization_id, "gupshup_apps"),
```

Comment:
can we make a key {org_id, "gupshup_app"} and put that in the cache, instead of all the apps in the case?
it will help to manage if a new app is added after updating the cache.